### PR TITLE
[new release] alcotest-lwt, alcotest and alcotest-async (1.0.1)

### PR DESCRIPTION
**Pull request by CraigFe**
Lwt-based helpers for Alcotest

- Project page: <a href="https://github.com/mirage/alcotest/">https://github.com/mirage/alcotest/</a>
- Documentation: <a href="https://mirage.github.io/alcotest/">https://mirage.github.io/alcotest/</a>

##### CHANGES:

- Add support for an `ALCOTEST_COLOR={auto,always,never}` environment variable
  to control the colorization of terminal output. (mirage/alcotest#209, @mjambon)
- Fix handling of exceptions in `Alcotest_{async,lwt}`. (mirage/alcotest#212,
  @CraigFe @talex5)

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Everything is green but somehow tests failed on `alcotest.1.0.1` on 4.05:
```
#=== ERROR while compiling alcotest.1.0.1 =====================================#
# context              2.0.6 | linux/x86_64 | ocaml-base-compiler.4.05.0 | pinned(https://github.com/mirage/alcotest/releases/download/1.0.1/alcotest-1.0.1.tbz)
# path                 ~/.opam/4.05/.opam-switch/build/alcotest.1.0.1
# command              ~/.opam/4.05/bin/dune runtest -p alcotest -j 24
# exit-code            1
# env-file             ~/.opam/log/alcotest-29-1872ec.env
# output-file          ~/.opam/log/alcotest-29-1872ec.out
### output ###
#       simple alias examples/runtest
# Testing Utils.
# This run has ID `4F73FCC2-8AA0-471F-97CF-BC44A20AFCF2`.
#  ...                string-case            0   Lower case.
[OK]                string-case            0   Lower case.
#  ...                string-case            1   Capitalization.
[OK]                string-case            1   Capitalization.
#  ...                string-concat          0   String mashing.
[OK]                string-concat          0   String mashing.
#  ...                list-concat            0   List mashing.
[OK]                list-concat            0   List mashing.
# The full test results are available in `/home/opam/.opam/4.05/.opam-switch/build/alcotest.1.0.1/_build/default/examples/_build/_tests/4F73FCC2-8AA0-471F-97CF-BC44A20AFCF2`.
# Test Successful in 0.000s. 4 tests run.
# ASSERT same string
# ASSERT same string
# ASSERT same string
# ASSERT same lists
# File "test/e2e/alcotest/failing/invalid_arg_in_test.expected", line 1, characters 0-0:
#          git (internal) (exit 1)
# (cd _build/default && /usr/bin/git diff --no-index --color=always -u test/e2e/alcotest/failing/invalid_arg_in_test.expected test/e2e/alcotest/failing/invalid_arg_in_test.processed)
# diff --git a/test/e2e/alcotest/failing/invalid_arg_in_test.expected b/test/e2e/alcotest/failing/invalid_arg_in_test.processed
# index 54d5fb4..c84d6f0 100644
# --- a/test/e2e/alcotest/failing/invalid_arg_in_test.expected
# +++ b/test/e2e/alcotest/failing/invalid_arg_in_test.processed
# @@ -1,11 +1,6 @@
# +invalid_arg_in_test.exe: internal error, uncaught exception:
# +                         Unix.Unix_error(Unix.EEXIST, "symlink", "/home/opam/.opam/4.05/.opam-switch/build/alcotest.1.0.1/_build/default/test/e2e/alcotest/failing/_build/_tests/latest")
# +                         
#  Testing suite-with-failures.
#  This run has ID `<uuid>`.
# - ...                test-a          0   Passing.
[OK]                test-a          0   Passing.
# - ...                test-a          1   Failing.
[FAIL]              test-a          1   Failing.
# - ...                test-b          0   Another pass.
[OK]                test-b          0   Another pass.
# --- test-a.001 [Failing.] Failed --
# -in `<build-context>/_build/_tests/<uuid>/test-a.001.output`:
# -[invalid] Failing test
# -
# -The full test results are available in `<build-context>/_build/_tests/<uuid>`.
# -1 error! in <test-duration>s. 3 tests run.
# +Fatal error: exception Alcotest__Core.Make(M).Test_error
#       floats alias examples/runtest
# Testing Float tests.
# This run has ID `3558FA77-FB97-4AD4-91EB-B972DCFF9E23`.
#  ...                Edge cases            0   NaN.
[OK]                Edge cases            0   NaN.
#  ...                Edge cases            1   ∞.
[OK]                Edge cases            1   ∞.
#  ...                Other floats          0   others.
[OK]                Other floats          0   others.
# The full test results are available in `/home/opam/.opam/4.05/.opam-switch/build/alcotest.1.0.1/_build/default/examples/_build/_tests/3558FA77-FB97-4AD4-91EB-B972DCFF9E23`.
# Test Successful in 0.000s. 3 tests run.
```
Could you look into it?

**Comment by kit-ty-kate** [comment]
Could you also return the fixes upstream?

**Comment by CraigFe** [comment]
> Could you also return the fixes upstream?

Done in https://github.com/mirage/alcotest/pull/216.

**Comment by CraigFe** [comment]
> Could you look into it?

I'm not seeing this failure in the CI; is it an intermittent failure?

**Comment by kit-ty-kate** [comment]
> I'm not seeing this failure in the CI; is it an intermittent failure?

It might be. Should I proceed to merging or is it worth more investigation based on the log above?

**Comment by CraigFe** [comment]
It's possible that it's a real bug, but I think very unlikely to be introduced by this release. I say go ahead and we can fix any issues in subsequent patch releases; this release already fixes a major issue with `1.0.0`.

**Comment by kit-ty-kate** [comment]
Ok, thanks!
